src/resources/Resource.ts

```diff
@@ -0,0 +1,36 @@
+import API from '../APICore';
+
+class Resource {
+    static baseUrl: string;
+
+    constructor(protected api: API) {}
```

Comment:
Kind of looks weird that you have `api` here and that it is only used in the *parent* classes.

Looks like a smell to me

Reply:
Its only used in the child classes (the ones that extends `Resource`). That way I don't have to define a constructor for each resource.